src/discovery/local-discovery.js

```diff
@@ -141,11 +152,16 @@ export class LocalDiscovery extends TypedEmitter {
       }
     }
 
-    const { remoteAddress } = socket
+    const { remoteAddress, remotePort } = socket
     if (!remoteAddress || !isPrivate(remoteAddress)) {
       socket.destroy(new Error('Invalid remoteAddress ' + remoteAddress))
       return
     }
+
+    // TODO: What if we already have an incoming connection from them?
+    const ipId = `${remoteAddress}:${remotePort}`
+    this.#connectionsFromIP.add(ipId)
+
```

Comment:
This won't work - the remote port is random and changes for each connection. e.g. run this and you'll see the port is different on each connection:

```js
import { once } from 'node:events'
import net from 'node:net'

const server = net.createServer()
server.listen(0, '0.0.0.0')

server.on('connection', (socket) => {
  console.log(`${socket.remoteAddress}:${socket.remotePort} connected`)
})

await once(server, 'listening')

net.connect({ port: server.address().port, host: '127.0.0.1' })
net.connect({ port: server.address().port, host: '127.0.0.1' })
```